### PR TITLE
Move HttpClient.request to StdLibExecution

### PR DESCRIPTION
**Pull request by OceanOak**
Changelog:

```
Stdlib
- Add HttpClient.requestFromWasm to Wasm/Libs
```

### CODE REVIEW

File: backend/src/StdLibExecution/Libs/HttpClient.fs [hunk 1]
```diff
@@ -4,18 +4,233 @@
...
+  // clients or just one, we use just one for efficiency.
+  // See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-7.0#alternatives-to-ihttpclientfactory
+  //
+  // Note that the number of sockets was verified manually, with:
+  // `sudo netstat -apn | grep _WAIT`
+  // TODO: I don't see where "the number of sockets" is actually configured?
+  // let private socketHandler : HttpMessageHandler =
+  //   new SocketsHttpHandler(
+  //     // Avoid DNS problems
+  //     PooledConnectionIdleTimeout = System.TimeSpan.FromMinutes 5.0,
```
Comment (StachuDotNet) [inline backend/src/StdLibExecution/Libs/HttpClient.fs:67]: first, we got errors around PooledConnectionIdleTimeout, so we decided to use a boring `new SocketsHttpHandler()`

File: backend/src/StdLibExecution/Libs/HttpClient.fs [hunk 2]
```diff
@@ -4,18 +4,233 @@
...
+
+  //     // Don't add a RequestId header for opentelemetry
+  //     ActivityHeadersPropagator = null,
+
+  //     // Users share the HttpClient, don't let them share cookies!
+  //     UseCookies = false
+  //   )
+
+  let private httpClient : HttpClient =
+    new HttpClient(
```
Comment (StachuDotNet) [inline backend/src/StdLibExecution/Libs/HttpClient.fs:94]: second, it turned out that by providing _any_ sockets handler to the `new HttpClient()`, it overrode the Blazor functionality of using the browser's `fetch` for the http client, so we reduced it to a more boring `new HttpClient()`

File: backend/src/Wasm/Libs/WASM.fs [hunk 3]
```diff
@@ -13,8 +13,186 @@ open LibExecution.StdLib.Shortcuts
...
+              Path = uri.AbsolutePath,
+              Query = uri.Query
+            )
+            |> string
+
+          use req =
+            new HttpRequestMessage(
+              httpRequest.method,
+              reqUri,
+              // Content = new ByteArrayContent(httpRequest.body),
```
Comment (OceanOak) [inline backend/src/Wasm/Libs/WASM.fs:90]: To check:
Uncommenting Content, I get a TypeError
ypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body. TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.

File: backend/src/Wasm/Libs/WASM.fs [hunk 4]
```diff
@@ -13,8 +13,186 @@ open LibExecution.StdLib.Shortcuts
...
+    }
+
+let headersType = TList(TTuple(TString, TString, []))
+
+type HeaderError =
+  | BadInput of string
+  | TypeMismatch of string
+
+let types : List<BuiltInType> =
+  [ { name = typ "WASM" "Response" 0
```
Comment (OceanOak) [inline backend/src/Wasm/Libs/WASM.fs:185]: Should this be HttpClient instead of WASM?

File: backend/src/Wasm/Libs/WASM.fs [hunk 5]
```diff
@@ -68,6 +246,115 @@ let fns : List<BuiltInFn> =
         | _ -> incorrectArgs ())
       sqlSpec = NotQueryable
       previewable = Impure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "WASM" "requestFromWasm" 0
```
Comment (OceanOak) [inline backend/src/Wasm/Libs/WASM.fs:252]: This function should probably be renamed?

**Comment by OceanOak** [comment]
Should these changes be in a new HttpClient file in Wasm/Libs?

File: backend/src/Wasm/Libs/WASM.fs (hunk 5 above)
Comment (pbiggar) [reply backend/src/Wasm/Libs/WASM.fs:252]: fyi we allow multiple modules now. I think you can do `name = fn' ["Wasm";"HttpClient";"Even";"More";"If";"Desired"] "request" 0`

**Pull request merged**